### PR TITLE
leocad: 21.03 -> 21.06 + library update + povray functionality

### PR DESCRIPTION
**Pull request by eliasp**
###### Motivation for this change
- bump leocad to its latest release
- update the parts library
- enable the povray rendering functionality

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/leocad/default.nix [hunk 1]
```diff
@@ -14,26 +15,34 @@ set the variable LEOCAD_LIB=/path/to/libs/ or use option -l /path/to/libs/
...
 
   buildInputs = [ zlib ];
 
+  propagatedBuildInputs = [ povray ];
+
+  patches = [ ./6e7dd2c7-povray.patch ];
+
+  postPatch = ''
+    substituteInPlace qt/lc_renderdialog.cpp --replace "@@NIX_POVRAY_PATH@@" "${povray}/bin/povray"
+  '';
```
Comment (sikmir) [inline pkgs/applications/graphics/leocad/default.nix:44]:
```suggestion
  patches = (substituteAll {
    src = ./6e7dd2c7-povray.patch;
    inherit povray;
  });
```

File: pkgs/applications/graphics/leocad/6e7dd2c7-povray.patch [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+diff --git a/common/lc_profile.cpp b/common/lc_profile.cpp
+index 1975b586..911c4fb0 100644
+--- a/common/lc_profile.cpp
++++ b/common/lc_profile.cpp
+@@ -132,7 +132,7 @@ static lcProfileEntry gProfileEntries[LC_NUM_PROFILE_KEYS] =
+ 	lcProfileEntry("HTML", "ImageWidth", 640),                                                 // LC_PROFILE_HTML_IMAGE_WIDTH
+ 	lcProfileEntry("HTML", "ImageHeight", 480),                                                // LC_PROFILE_HTML_IMAGE_HEIGHT
+ 
+-	lcProfileEntry("POVRay", "Path", "/usr/bin/povray"),                                       // LC_PROFILE_POVRAY_PATH
++	lcProfileEntry("POVRay", "Path", "@@NIX_POVRAY_PATH@@"),                                   // LC_PROFILE_POVRAY_PATH
```
Comment (sikmir) [inline pkgs/applications/graphics/leocad/6e7dd2c7-povray.patch:39]:
```suggestion
+	lcProfileEntry("POVRay", "Path", "@povray@/bin/povray"),                                   // LC_PROFILE_POVRAY_PATH
```

File: pkgs/applications/graphics/leocad/6e7dd2c7-povray.patch [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+diff --git a/qt/lc_renderdialog.cpp b/qt/lc_renderdialog.cpp
+index bd8a9102..96794738 100644
+--- a/qt/lc_renderdialog.cpp
++++ b/qt/lc_renderdialog.cpp
+@@ -184,7 +184,7 @@ void lcRenderDialog::on_RenderButton_clicked()
+ #endif
+ 
+ #ifdef Q_OS_LINUX
+-	POVRayPath = lcGetProfileString(LC_PROFILE_POVRAY_PATH);
++	POVRayPath = QDir::cleanPath(QLatin1String("@@NIX_POVRAY_PATH@@"));
```
Comment (sikmir) [inline pkgs/applications/graphics/leocad/6e7dd2c7-povray.patch:66]:
```suggestion
+	POVRayPath = QDir::cleanPath(QLatin1String("@povray@/bin/povray"));
```

File: pkgs/applications/graphics/leocad/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/graphics/leocad/default.nix:44]: Please also remove the hash from the patch file

**Comment by eliasp** [comment]
Thanks for the review, @sikmir and @SuperSandro2000 - I updated the PR accordingly.

File: pkgs/applications/graphics/leocad/default.nix [hunk 4]
```diff
@@ -14,26 +16,33 @@ set the variable LEOCAD_LIB=/path/to/libs/ or use option -l /path/to/libs/
...
   nativeBuildInputs = [ qmake qttools ];
 
   buildInputs = [ zlib ];
 
+  propagatedBuildInputs = [ povray ];
+
+  patches = (substituteAll {
+    src = ./povray.patch;
+    inherit povray;
+  });
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/leocad/default.nix:44]:
```suggestion
  patches = [
    (substituteAll {
      src = ./povray.patch;
      inherit povray;
    })
  ];
```

**Pull request merged**